src/x509.c

```diff
@@ -7202,6 +7231,433 @@ WOLFSSL_X509_CRL* wolfSSL_d2i_X509_CRL(WOLFSSL_X509_CRL** crl,
+                if (X509RevokedPrintSerial(bio, revoked, indent + 4) != WOLFSSL_SUCCESS) {
+                    return WOLFSSL_FAILURE;
+                }
+            }
+        #ifndef NO_ASN_TIME
+             XSNPRINTF(tmp, MAX_WIDTH, "%*s%s", indent + 8, "",
+                "Revocation Date: ");
+
+            if (wolfSSL_BIO_write(bio, tmp, (int)XSTRLEN(tmp)) <= 0) {
+                return WOLFSSL_FAILURE;
+            }
+
+            if (revoked->revDate[0] != 0) {
+                if (GetTimeString(revoked->revDate, ASN_UTC_TIME,
+                    tmp, MAX_WIDTH) != WOLFSSL_SUCCESS) {
+                    if (GetTimeString(revoked->revDate, ASN_GENERALIZED_TIME,
+                    tmp, MAX_WIDTH) != WOLFSSL_SUCCESS) {
+                        WOLFSSL_MSG("Error getting revocation date");
+                        return WOLFSSL_FAILURE;
+                    }
+                }
+            }
+            else {
+                XSTRNCPY(tmp, "Not Set", MAX_WIDTH-1);
+            }
+            tmp[MAX_WIDTH - 1] = '\0'; /* make sure null terminated */
+            if (wolfSSL_BIO_write(bio, tmp, (int)XSTRLEN(tmp)) <= 0) {
+                return WOLFSSL_FAILURE;
+            }
+
+            if (wolfSSL_BIO_write(bio, "\n", (int)XSTRLEN("\n")) <= 0) {
+                return WOLFSSL_FAILURE;
+            }
+        #endif
+            revoked = revoked->next;
+        }
+    }
+    else {
+        if (wolfSSL_BIO_write(bio, "No Revoked Certificates.\n",
+                       (int)XSTRLEN("No Revoked Certificates.\n")) <= 0) {
+            return WOLFSSL_FAILURE;
+        }
+    }
+
+    return WOLFSSL_SUCCESS;
+}
+
+#if !defined(NO_BIO) && defined(XSNPRINTF)
+int wolfSSL_X509_CRL_print(WOLFSSL_BIO* bio, WOLFSSL_X509_CRL* crl)
+{
+    if (bio == NULL || crl == NULL || crl->crlList == NULL) {
+        return WOLFSSL_FAILURE;
+    }
+
+    if (wolfSSL_BIO_write(bio, "Certificate Revocation List (CRL):\n",
+                  (int)XSTRLEN("Certificate Revocation List (CRL):\n")) <= 0) {
+            return WOLFSSL_FAILURE;
+    }
+
+    /* print version */
+    if (X509CRLPrintVersion(bio, crl, 8) != WOLFSSL_SUCCESS) {
+        return WOLFSSL_FAILURE;
+    }
+
+    /* print signature algo */
+    {
+        int oid;
+        const char* sig;
+
+        if ((oid = wolfSSL_X509_CRL_get_signature_type(crl)) <= 0) {
+            WOLFSSL_MSG("Error getting x509 signature type");
+            return WOLFSSL_FAILURE;
+        }
+        if (wolfSSL_BIO_write(bio, "        Signature Algorithm: ",
+                      (int)XSTRLEN("        Signature Algorithm: ")) <= 0) {
+            return WOLFSSL_FAILURE;
+        }
+
+        sig = GetSigName(oid);
+        if (wolfSSL_BIO_write(bio, sig, (int)XSTRLEN(sig)) <= 0) {
+            return WOLFSSL_FAILURE;
+        }
+        if (wolfSSL_BIO_write(bio, "\n", (int)XSTRLEN("\n")) <= 0) {
+            return WOLFSSL_FAILURE;
+        }
+    }
+
+    /* print issuer */
+     {
```

Comment:
use

```
+    /* print issuer */
+    if (X509PrintName(bio, wolfSSL_CRL_get_issuer_name(crl), issuType, 8)
+            != WOLFSSL_SUCCESS) {
+        return WOLFSSL_FAILURE;
+    }
```

After changing X509PrintName to take in a WOLFSSL_X509_NAME.

Reply:
Done